More than one tag on a path breaks server code generation

## Problem statement

A swagger definition that includes more than one tag in a path's tag definitions generates bad code. It appears to nest a second `operations` directory within the first.

## Swagger specification
```yaml
swagger: '2.0'
info:
  title: Multi tag API
  description: multi tag api
  version: '0.0.1'

# the domain of the service
host: api.multi.tags
schemes: [ https ]
produces: [ application/json ]
consumes: [ application/json ]

basePath: /v1 # Prefixed to all paths
paths:
  /multi:
    get:
      summary: A multitag route
      tags:
        - onetag
        - twotag
      responses:
        default:
          description: Unexpected error
          schema:
            type: object
            properties:
              type:
                type: string
                description: The error type
              message:
                type: array
                description: Explanation of the errors
                items: { type: string }
```

## Steps to reproduce
`swagger generate server` with the above, and attempt to go build.

You'll get the following error:
```
restapi/operations/multi_tag_api.go:54: undefined: GetMultiHandler
```

## Environment
swagger version: dev
go version: 1.7.4
OS: macOS

Im seeing the operations/operations folder created. Tried to compile my generated server but getting compile errors. 
2017/09/05 14:22:18 building a plan for generation
2017/09/05 14:22:18 planning definitions
2017/09/05 14:22:18 planning operations
2017/09/05 14:22:18 grouping operations into packages
2017/09/05 14:22:18 planning meta data and facades
2017/09/05 14:22:18 rendering 3 models
2017/09/05 14:22:18 rendering 1 templates for model uuid
2017/09/05 14:22:18 name field uuid
2017/09/05 14:22:18 package field models
2017/09/05 14:22:18 creating "uuid.go" in "/home/prasadj/workspace/main6/.go/src/nusights/server/apis/models" as definition
2017/09/05 14:22:18 rendering 1 templates for model error
2017/09/05 14:22:18 name field error
2017/09/05 14:22:18 package field models
2017/09/05 14:22:18 creating "error.go" in "/home/prasadj/workspace/main6/.go/src/nusights/server/apis/models" as definition
2017/09/05 14:22:18 rendering 1 templates for model success
2017/09/05 14:22:18 name field success
2017/09/05 14:22:18 package field models
2017/09/05 14:22:18 creating "success.go" in "/home/prasadj/workspace/main6/.go/src/nusights/server/apis/models" as definition
2017/09/05 14:22:18 rendering 1 operation groups (tags)
2017/09/05 14:22:18 rendering 1 operations for operations
2017/09/05 14:22:18 rendering 4 templates for operation nusights-rest-api
2017/09/05 14:22:18 name field fileupload
2017/09/05 14:22:18 package field operations
2017/09/05 14:22:18 creating "fileupload_parameters.go" in "/home/prasadj/workspace/main6/.go/src/nusights/server/apis/restapi/operations/operations" as parameters
2017/09/05 14:22:18 name field fileupload
2017/09/05 14:22:18 package field operations
2017/09/05 14:22:18 creating "fileupload_urlbuilder.go" in "/home/prasadj/workspace/main6/.go/src/nusights/server/apis/restapi/operations/operations" as urlbuilder
2017/09/05 14:22:18 name field fileupload
2017/09/05 14:22:18 package field operations
2017/09/05 14:22:18 creating "fileupload_responses.go" in "/home/prasadj/workspace/main6/.go/src/nusights/server/apis/restapi/operations/operations" as responses
2017/09/05 14:22:18 name field fileupload
2017/09/05 14:22:18 package field operations
2017/09/05 14:22:18 creating "fileupload.go" in "/home/prasadj/workspace/main6/.go/src/nusights/server/apis/restapi/operations/operations" as handler
2017/09/05 14:22:18 rendering support
2017/09/05 14:22:18 rendering 6 templates for application NusightsRest
2017/09/05 14:22:18 name field NusightsRest
2017/09/05 14:22:18 package field operations
2017/09/05 14:22:18 skipping /home/prasadj/workspace/main6/.go/src/nusights/server/apis/restapi/configure_nusights_rest.go because it already exists
2017/09/05 14:22:18 name field NusightsRest
2017/09/05 14:22:18 package field operations
2017/09/05 14:22:18 creating "main.go" in "/home/prasadj/workspace/main6/.go/src/nusights/server/apis/cmd/nusights-rest-server" as main
2017/09/05 14:22:18 name field NusightsRest
2017/09/05 14:22:18 package field operations
2017/09/05 14:22:18 creating "embedded_spec.go" in "/home/prasadj/workspace/main6/.go/src/nusights/server/apis/restapi" as embedded_spec
2017/09/05 14:22:18 name field NusightsRest
2017/09/05 14:22:18 package field operations
2017/09/05 14:22:18 creating "server.go" in "/home/prasadj/workspace/main6/.go/src/nusights/server/apis/restapi" as server
2017/09/05 14:22:18 name field NusightsRest
2017/09/05 14:22:18 package field operations
2017/09/05 14:22:18 creating "nusights_rest_api.go" in "/home/prasadj/workspace/main6/.go/src/nusights/server/apis/restapi/operations" as builder
2017/09/05 14:22:18 name field NusightsRest
2017/09/05 14:22:18 package field operations
2017/09/05 14:22:18 creating "doc.go" in "/home/prasadj/workspace/main6/.go/src/nusights/server/apis/restapi" as doc
Generation completed!
Invoking go build for nusights_server_apis_cmd_nusights-rest-server
# nusights/server/apis/restapi/operations
.go/src/nusights/server/apis/restapi/operations/nusights_rest_api.go:71: undefined: FileuploadHandler
make[3]: *** [nusights/go/nusights/server/apis/cmd/nusights-rest-server/CMakeFiles/nusights_server_apis_cmd_nusights-rest-server] Error 2

--- a/fixtures/bugs/931/.gitignore
+++ b/fixtures/bugs/931/.gitignore
@@ -1,0 +1,3 @@
+gen-*
+!gen-fixtures.sh
+*.log

--- a/fixtures/bugs/931/fixture-931.yaml
+++ b/fixtures/bugs/931/fixture-931.yaml
@@ -1,0 +1,33 @@
+swagger: '2.0'
+info:
+  title: Multi tag API
+  description: multi tag api
+  version: '0.0.1'
+
+# the domain of the service
+host: api.multi.tags
+schemes: [ https ]
+produces: [ application/json ]
+consumes: [ application/json ]
+
+basePath: /v1 # Prefixed to all paths
+paths:
+  /multi:
+    get:
+      summary: A multitag route
+      tags:
+        - onetag
+        - twotag
+      responses:
+        default:
+          description: Unexpected error
+          schema:
+            type: object
+            properties:
+              type:
+                type: string
+                description: The error type
+              message:
+                type: array
+                description: Explanation of the errors
+                items: { type: string }

--- a/fixtures/bugs/931/gen-fixtures.sh
+++ b/fixtures/bugs/931/gen-fixtures.sh
@@ -1,0 +1,74 @@
+#! /bin/bash 
+if [[ ${1} == "--clean" ]] ; then
+    clean=1
+fi
+continueOnError=
+# A small utility to build fixture servers
+# Fixtures with models only
+testcases="${testcases} fixture-931.yaml"
+for opts in  "" ; do
+for testcase in ${testcases} ; do
+    grep -q discriminator ${testcase}
+    discriminated=$?
+    if [[ ${discriminated} -eq 0 && ${opts} == "--with-expand" ]] ; then
+        echo "Skipped ${testcase} with ${opts}: discriminator not supported with ${opts}"
+        continue
+    fi
+    if [[ ${testcase} == "../1479/fixture-1479-part.yaml" && ${opts} == "--with-expand" ]] ; then
+        echo "Skipped ${testcase} with ${opts}: known issue with enum in anonymous allOf not validated. See you next PR"
+        continue
+    fi
+
+    spec=${testcase}
+    testcase=`basename ${testcase}`
+    if [[ -z ${opts} ]]; then
+        target=./gen-${testcase%.*}-flatten
+    else
+        target=./gen-${testcase%.*}-expand
+    fi
+    serverName="codegensrv"
+    rm -rf ${target}
+    mkdir ${target}
+    echo "Model generation for ${spec} with opts=${opts}"
+    serverName="nrcodegen"
+    swagger generate server --skip-validation ${opts} --spec ${spec} --target ${target} --name=${serverName} 1>${testcase%.*}.log 2>&1
+    # 1>x.log 2>&1
+    #
+    if [[ $? != 0 ]] ; then
+        echo "Generation failed for ${spec}"
+        if [[ ! -z ${continueOnError} ]] ; then
+            failures="${failures} codegen:${spec}"
+            continue
+        else
+            exit 1
+        fi
+    fi
+    echo "${spec}: Generation OK"
+    if [[ ! -d ${target}/models ]] ; then
+        echo "No model in this spec! Continue building server"
+    fi
+    if [[ -d ${target}/cmd/${serverName}"-server" ]] ; then
+        (cd ${target}/cmd/${serverName}"-server"; go build)
+        #(cd ${target}/models; go build)
+        if [[ $? != 0 ]] ; then
+            echo "Build failed for ${spec}"
+            if [[ ! -z ${continueOnError} ]] ; then
+                failures="${failures} build:${spec}"
+                continue
+            else
+                exit 1
+            fi
+        fi
+        echo "${spec}: Build OK"
+        if [[ -n ${clean} ]] ; then 
+             rm -rf ${target}
+        fi
+    fi
+done
+done
+if [[ ! -z ${failures} ]] ; then 
+    echo ${failures}|tr ' ' '\n'
+else
+    echo "No failures"
+fi
+exit

--- a/hack/codegen-nonreg.sh
+++ b/hack/codegen-nonreg.sh
@@ -153,6 +153,7 @@ specdir=${specdir}" ../fixtures/bugs/1487"
 specdir=${specdir}" ../fixtures/bugs/1571"
 specdir=${specdir}" ../fixtures/bugs/957"
 specdir=${specdir}" ../fixtures/bugs/1614"
+specdir=${specdir}" ../fixtures/bugs/931"
 gendir=./tmp-gen
 rm -rf ${gendir}
 

